src/SFML/Main/MainWin32.cpp

```diff
@@ -41,6 +41,11 @@
 
 #include <windows.h>
 
+// Inform the Nvidia driver that this SFML application should
+// make use of the more powerful discrete GPU
+extern "C" {
+__declspec(dllexport) DWORD NvOptimusEnablement = 0x00000001;
```

Comment:
I can't think of any situation where someone would not want the driver to automatically select the discrete GPU. Note that this only has an effect if the user wants automatic detection, they can still override which adapter is used in their Nvidia control panel. This is mainly used to address the issue of OpenGL not having any other way of selecting the adapter to use, unlike Direct3D from what I understand. For games/applications for which an Nvidia optimus profile already exists, the driver will select the discrete GPU if set to automatic, so I think we are only doing the application developers a favour by providing an easy way to allow for automatic detection when using SFML.